Add autosectionlabel_full_reference configuration variable (#13076)
sphinx-doc/sphinx: doc/usage/extensions/autosectionlabel.rst

section names are used in whole of document, any one is used for a target by
default. The ``autosectionlabel_prefix_document`` configuration variable can be
used to make headings which appear multiple times but in different documents

> unique. Also the ``autosectionlabel_full_reference`` configuration variable can
> help ensure the generated refereences are unique across a document with similar
> section names at different levels.

Reviewer: ```suggestion
unique. 

Use the :conf:`autosectionlabel_full_reference` configuration variable
to guarantee that the generated references are unique across a document
with similar section names at different levels.
```

section names are used in whole of document, any one is used for a target by
default. The ``autosectionlabel_prefix_document`` configuration variable can be
used to make headings which appear multiple times but in different documents
unique.

> Use the :conf:`autosectionlabel_full_reference` configuration variable

Reviewer: ```suggestion
Use the :confval:`autosectionlabel_full_reference` configuration variable
```

   only for top level sections, and deeper sections are not labeled.  It
   defaults to ``None`` (i.e. all sections are labeled).

.. confval:: autosectionlabel_full_reference
   :type: :code-py:`bool`
   :default: :code-py:`False`

   True to make each section label include all the parent sections separated by

>    a colon. For example::
> 
>       Title
>       =====
> 
>       Section
>       -------
> 
>       Sub Section
>       ~~~~~~~~~~~

Reviewer: ```suggestion
   a colon. For instance:
   
   .. code-block:: rst

      Title
      =====

      Section
      -------

      Sub Section
      ~~~~~~~~~~~
```

>    Then the reference of the third level section is
>    ``index:Title:Section:Sub Section``.

Reviewer: ```suggestion
   The reference of the third level section ("Sub Section") will be
   ``index:Title:Section:Sub Section``.
```
